samples/drivers/espi/src/main.c

```diff
@@ -0,0 +1,339 @@
+#define PORTC_LABEL            "PORTC"
+#define BOARD_PWRGD_PIN        0
+#define WAKE_SCI_PIN           1
+#define ESPI_MASTER_INIT_PIN   2
+#define SYSTEM_UP_PIN          3
+#define ESPI_MASTER_RUN_PIN    4
+#define BOARD_RUN_PIN          5
+#endif
+
+
+static struct device *gpio_devA;
+static struct device *gpio_devB;
+static struct device *gpio_devC;
+static struct device *espi_dev;
+
+static struct espi_callback espi_bus_cb;
+static struct espi_callback vw_rdy_cb;
+static struct espi_callback vw_cb;
+static struct espi_callback p80_cb;
+
+static void espi_reset_handler(struct device *dev,
+				struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_RESET) {
+		printk("\neSPI BUS reset %d", event.evt_data);
+	}
+}
+
+static void ch_rdy_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_EVENT_CHANNEL_READY) {
+		if (event.evt_details == ESPI_CHANNEL_VWIRE) {
+			printk("\nVW channel is ready\n");
+		}
+	}
+}
+
+static void vwire_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_EVENT_VWIRE_RECEIVED) {
+		printk("\n%d%d\n", event.evt_details, event.evt_data);
+	}
+}
+
+static void port80_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_PERIPHERAL_NOTIFICATION) {
+		printk("\n%s %x\n", __func__, event.evt_data);
+	}
+}
+
+int espi_init(void)
+{
+	int ret;
+	/* Indicate to eSPI master only single line, 20MHz frequency and
+	 * Logical channel 0 are supported
+	 */
+	struct espi_cfg cfg = {
+		ESPI_IO_MODE_SINGLE_LINE,
+		ESPI_CHANNEL_VWIRE,
+		20,
+		};
+
+	ret = espi_config(espi_dev, &cfg);
+	if (ret) {
+		printk("Failed to configure eSPI slave! error (%d)\n", ret);
+	} else {
+		printk("eSPI slave configured succesfully!\n");
+	}
+
+	printk("eSPI test - callbacks initialization... ");
+	espi_init_callback(&espi_bus_cb, espi_reset_handler,
+				ESPI_BUS_RESET);
+	espi_init_callback(&vw_rdy_cb, ch_rdy_handler,
+				ESPI_BUS_EVENT_CHANNEL_READY);
+	espi_init_callback(&vw_cb, vwire_handler,
+				ESPI_BUS_EVENT_VWIRE_RECEIVED);
+	espi_init_callback(&p80_cb, port80_handler,
+				ESPI_BUS_PERIPHERAL_NOTIFICATION);
+	printk("complete\n");
+
+	printk("eSPI test - callbacks registration... ");
+	espi_add_callback(espi_dev, &espi_bus_cb);
+	espi_add_callback(espi_dev, &vw_rdy_cb);
+	espi_add_callback(espi_dev, &vw_cb);
+	espi_add_callback(espi_dev, &p80_cb);
+	printk("complete\n");
+	return ret;
+}
+
+int io_init(void)
+{
+	int ret = 0;
+
+#ifdef CONFIG_GPIO_XEC
+	printk("eSPI test - I/O initialization...");
+	GPIO_CTRL_REGS->CTRL_0012 = MCHP_GPIO_CTRL_MUX_F0;
+	GPIO_CTRL_REGS->CTRL_0054 = MCHP_GPIO_CTRL_MUX_F0;
+	GPIO_CTRL_REGS->CTRL_0057 = MCHP_GPIO_CTRL_MUX_F0;
+	GPIO_CTRL_REGS->CTRL_0115 = MCHP_GPIO_CTRL_MUX_F0;
+#endif
+	ret = gpio_pin_configure(gpio_devA, BOARD_PWRGD_PIN, GPIO_DIR_IN);
+	if (ret)
```

Comment:
always have '{ ... }' even for one liner statement in if/for/while. Apply that everywhere.